persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/message/MessageDAOImpl.kt

```diff
@@ -184,6 +190,31 @@ class MessageDAOImpl(
         }
     }
 
+    private fun updateAssetMessage(message: MessageEntity) {
+        val assetMessageContent = message.content as MessageEntityContent.Asset
+        queries.updateAssetKeys(
+            assetId = assetMessageContent.assetId,
+            assetOtrKey = assetMessageContent.assetOtrKey,
+            assetSha256 = assetMessageContent.assetSha256Key,
+            visibility = message.visibility
+        )
+    }
+
+    /**
+     * Returns true if the [message] is an asset message that is already in the DB and any of its decryption keys are null/empty. This means
+     * that this asset message that is in the DB was only a preview message with valid metadata but no valid keys (Web clients send 2
+     * separated messages). Therefore it still needs to be updated with the valid keys in order to be displayed.
+     */
+    private fun isAssetMessageUpdate(message: MessageEntity): Boolean {
+        return if (message is MessageEntity.Regular && message.content is MessageEntityContent.Asset) {
+            queries.selectById(message.id, message.conversationId).executeAsList().firstOrNull()?.let {
+                if (it.assetId.isNullOrEmpty() || it.assetOtrKey == null || it.assetSha256 == null) {
+                    return true
+                } else false
+            } ?: return false
+        } else false
+    }
```

Comment:
Maybe a bit more readable? Trying to reduce having too many true/false statements side by side.

```suggestion
     * separated messages). Therefore, it still needs to be updated with the valid keys in order to be displayed.
     */
    private fun isAssetMessageUpdate(message: MessageEntity): Boolean {
        val isAssetMessage = message is MessageEntity.Regular && message.content is MessageEntityContent.Asset
        val currentMessageHasMissingAssetInformation = if (isAssetMessage) {
            queries.selectById(message.id, message.conversationId).executeAsList().firstOrNull()?.let {
                it.assetId.isNullOrEmpty() || it.assetOtrKey == null || it.assetSha256 == null
            } ?: false
        } else false
        return currentMessageHasMissingAssetInformation
    }
```